Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/Merchant/Overlay.hs

```diff
@@ -43,3 +44,11 @@ cacheOverlay pn = do
 
 makeMerchantIdPNKeyLangaugeUdf :: Id Merchant -> Text -> Language -> Maybe Text -> Text
 makeMerchantIdPNKeyLangaugeUdf id pnKey language udf1 = "CachedQueries:Overlay:MerchantId-" <> id.getId <> ":PNKey-" <> show pnKey <> ":ln-" <> show language <> ":udf1-" <> show udf1
+
+------------------------------------------------------
+
+clearCacheByOverlayKeyAndMerchantId :: Hedis.HedisFlow m r => Id Merchant -> Text -> m ()
+clearCacheByOverlayKeyAndMerchantId merchantId overlayKey = Hedis.delByPattern $ makeOverlayPatternKey merchantId overlayKey
+
+makeOverlayPatternKey :: Id Merchant -> Text -> Text
+makeOverlayPatternKey merchantId overlayKey = "CachedQueries:Overlay:MerchantId-" <> merchantId.getId <> ":PNKey-" <> show overlayKey <> ":*"
```

Comment:
We should not use `*` at all for Redis. As this is very heavy operation and can bring down Redis.